### PR TITLE
remove unsupported functionality from cluster-api provider

### PR DESCRIPTION
**Pull request by elmiko**
#### Which component this PR applies to?

cluster-autoscaler

#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

this change removes the code for the `Labels` and `Taints` interface functions of the clusterapi provider when scaling from zero. The body of these functions was added erronesouly and the Cluster API community is still deciding on how these values will be expose to the autoscaler.

also updates the tests and readme to be more clear about the usage of labels and taints when scaling from zero.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cluster-autoscaler/cloudprovider/clusterapi/clusterapi_nodegroup_test.go [hunk 1]
```diff
@@ -1361,39 +1340,29 @@ func TestNodeGroupTemplateNodeInfo(t *testing.T) {
 				expectedErr:  nil,
 				nodeLabels: map[string]string{
 					"kubernetes.io/os":                 "windows",
+					"beta.kubernetes.io/os":            "windows",
 					"kubernetes.io/arch":               "arm64",
+					"beta.kubernetes.io/arch":          "arm64",
 					"node.kubernetes.io/instance-type": "instance1",
-					"anotherLabel":                     "nodeValue", // This should not be copied as it is not a well known label
-				},
-				nodegroupLabels: map[string]string{
```
Comment (asm582) [inline cluster-autoscaler/cloudprovider/clusterapi/clusterapi_nodegroup_test.go:1368]: is it the case that nodegroup labels will be added to node(s) under the `.metadata/labels` level?

**Review by mwielgus — APPROVED** [review]
/lgtm
/approve

File: cluster-autoscaler/cloudprovider/clusterapi/clusterapi_nodegroup_test.go (hunk 1 above)
Comment (elmiko) [reply cluster-autoscaler/cloudprovider/clusterapi/clusterapi_nodegroup_test.go:1368]: yes, when the node is created the labels will be added. but, we currently don't have a mechanism from the cluster-api side about where to find those labels, this issue is related https://github.com/kubernetes-sigs/cluster-api/issues/7006
